### PR TITLE
Refuse inf observations before the behavior model turns 0*inf into NaN

### PR DESCRIPTION
**Pull request by Kayvan-Zahiri**
The linear behavior model does `logit_error * x`. Inf observation makes softmax NaN, then a silent feature is `0 * inf` = NaN, and every weight, bias, and NLL EMA stays poisoned.

Hold the previous finite state when the observation or proposed weights/diagnostics are non-finite.

Failing-test-first: inf observation wrote NaN into every weight on main. `tests/test_behavior_model.py`: 20 passed.

Independent of #81 (RLS reward model).

No Slop run receipt: this session is Cursor Grok, not Codex `gpt-5.6-sol` or Claude Code `claude-fable-5`.

Made with [Cursor](https://cursor.com)

### CODE REVIEW

**Review by Prabhat1308 — APPROVED** [review]
Independent review of head `dd8939c3130c700278f3b2643103b0d66ccdab59` against `main` `da8b86c`, under the `review-asi-contributions` skill. Head tree verified as current `main` plus exactly `alberta_framework/core/behavior_model.py` + `tests/test_behavior_model.py`; automated adversarial scan clean (no dependency, CI, hook, or lockfile changes; no assertions or raises removed anywhere in the fleet).

**Reproduced:** `tests/test_behavior_model.py` at head — **20 passed in 1.70s**; repo-wide Ruff: All checks passed!; strict mypy: Success: no issues found in 159 source files; `git diff --check`: clean. Red phase with `alberta_framework/core/behavior_model.py` reverted to `main`, PR tests kept: **1 failed, 19 passed in 1.61s** (1 new regression test(s) fail), confirming failing-test-first.

**Design note (non-blocking):** beyond refusing non-finite inputs, the `proposed`-finiteness clause silently holds the previous state on *any* non-finite update — including overflow from finite-but-extreme dynamics — converting visible divergence into silent stalling. This matches the direction of the merged #43 family, but the hold-vs-surface contract for measured learners deserves an explicit maintainer position.

Process context, stated factually: this PR is part of a 27-PR same-day fleet by one author in the same non-finite-guard class, with honest Cursor authorship disclosure, no measured-run receipt, and no linked pre-registration issue; receipts cannot create score and their absence does not gate correctness — flagged only so the maintainer weighs the fleet, not just each PR.

Recommendation: **accept**.

```slop-review
{"schemaVersion": "1", "projectId": "asi", "artifactUrl": "https://github.com/elizaOS/asi/pull/82", "headSha": "dd8939c3130c700278f3b2643103b0d66ccdab59", "recommendation": "accept", "reproduced": true, "securityRisk": "none", "duplicateRisk": "none", "usefulArtifacts": ["non-finite guard in alberta_framework/core/behavior_model.py with failing-test-first regression test(s), reproduced red on main (1 failed, 19 passed in 1.61s) and green at head (20 passed in 1.70s)"], "commands": [".venv/bin/python -m pytest tests/test_behavior_model.py -q -o addopts=\"\"", ".venv/bin/python -m ruff check .", ".venv/bin/python -m mypy", "git diff --check"], "evidenceUrls": ["https://github.com/elizaOS/asi/pull/82"], "summary": "Head verified as main plus the stated two-file diff; red/green reproduced (1 regression test(s) red on main, 20 passed in 1.70s at head). Part of a 27-PR same-day non-finite-guard fleet, disclosed Cursor-authored, no receipts, no linked issues."}
```
